python/gui/MaaS/static/common/js/domain.js

```diff
@@ -0,0 +1,206 @@
+
+function titleCase(str) {
+    return str.replaceAll("_", " ").toLowerCase().split(' ').map(function(word) {
+        return word.replace(word[0], word[0].toUpperCase());
+    }).join(' ');
+}
+
+function loadFabricNames() {
```

Comment:
This would be a great place to get used to the `fetch` API. `$.ajax` does the job, but if you want to experiment a little, this is a great way to get used to async javascript in a small controlled manner.

Reply:
This derived directly from a similar function in _map.js_.  I don't really want to perfect the implementation at this stage, especially while trying to apply something considerably different (that I'm already not especially familiar with).  I'm going to leave this alone unless you feel like there is a really compelling reason to try to do this now.

Reply:
Hey, I wrote that _before_ I knew how to use `fetch`!

>  I don't really want to perfect the implementation at this stage, especially while trying to apply something considerably different (that I'm already not especially familiar with)

Great decision. Can you add a @TODO near it or a ticket for converting traditional ajax calls to the `fetch` api? Chances of addressing the TODO are probably low, but at least it'll be in there as low hanging fruit for anyone that might be interested.